server/src/main/java/com/google/musicanalysis/site/OAuthCallbackServlet.java

```diff
@@ -85,5 +87,9 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res)
 
     res.setContentType("text/html");
     res.getWriter().printf("<h1>the access token for %s is %s</h1>", getServiceName(), accessToken);
+    
+    // store access token in a session as youtube_access_token or spotify_access_token
+    HttpSession session = req.getSession();
+    session.setAttribute(getServiceName() + "_access_token", accessToken.toString());
```

Comment:
Nick will tell you to use constants instead of doing a concatenation here (he calls it "magic"). See `getSessionServiceKey()`, make a similar method like `getSessionTokenKey()` that just returns `"oauth-token-youtube"` or `"oauth-token-spotify"`.